include/aspect/stokes_matrix_free.h

```diff
@@ -57,6 +57,73 @@ namespace aspect
    */
   namespace MatrixFreeStokesOperators
   {
+
+    /**
+     * This struct stores the data for the current linear operator that is requried to perform
+     * matrix-vector products.
+     */
+    template <int dim, typename number>
+    struct OperatorCellData
+    {
+      /**
+       * Information on the compressibility of the flow.
+       */
+      bool is_compressible;
+
+      /**
+       * Pressure scaling constant.
+       */
+      double pressure_scaling;
+
+      /**
+       * If true, Newton terms are part of the operator.
+       */
+      bool enable_newton_derivatives;
+
+      /**
+       * Symmetrize the Newton system when it's true (i.e., the stabilization is symmetric or SPD).
+       */
+      bool symmetrize_newton_system;
+
+      /**
+       * Table which stores viscosity values for each cell.
+       */
+      Table<2, VectorizedArray<number>> viscosity;
+
+      /**
+       * Table which stores the product of viscosity derivative with respect to pressure
+       * and newton derivative scaling factor.
+       */
+      Table<2, VectorizedArray<number>> viscosity_derivative_wrt_pressure_table;
+
+      /**
+       * Table which stores the strain rate for each cell.
+       */
+      Table<2, SymmetricTensor<2, dim, VectorizedArray<number>>>
+      strain_rate_table;
+
+      /**
+       * Table which stores the product of the following three variables:
+       * viscosity derivative with respect to strain rate,
+       * newton derivative scaling factor, and alpha. Here alpha is the spd factor when the stabilization is PD or SPD,
+       * otherwise, it is 1.
+       */
+      Table<2, SymmetricTensor<2, dim, VectorizedArray<number>>>
+      viscosity_derivative_wrt_strain_rate_table;
```

Comment:
Can you name the variable to reflect that it contains this product? Right now it looks like it's just one of the three factors. Maybe something like `Newton_derivative_factor_table` if you don't want to put all of the three factors into the name.